### ISSUE TITLE
xxx does not support thin blocks, disconnecting

### ISSUE THREAD

**Issue by KlausT** [bitcoinxt/bitcoinxt#153]
Is this intended behavior when I don't use any special options?

> '/Satoshi:0.12.1/'` - peer=8 does not support thin blocks, disconnecting
> '/Classic:1.1.0/' - peer=11 does not support thin blocks, disconnecting

0.11.x and older clients are not being disconnected, as far as I can see

**Comment by dagurval** [comment]
At around the time Bitcoin XT started doing bloom thin blocks, Core changed the standard behaviour of merkleblock requests and broke compatibility with our thin block implementation. Classic also has this one liner change. More discussion here: https://github.com/bitcoinclassic/bitcoinclassic/pull/125

For _outgoing_ connections we avoid peers that don't support thin blocks. This means core > 0.12 and derivatives. We still _accept incoming_ connections from any peers and download full blocks from them (unless configured not to).

For now, this means outgoing connections to:
- Anyone that signals support for xthin blocks (XT, BU and hopefully soon Classic)
- Older XT nodes
- Bitcoin Core <= 0.11 nodes

So in other words, this is intended behaviour.

**Comment by NanoAkron** [comment]
I find any behaviour that segregates peers to be a bit distasteful. This is meant to be 'peer-to-peer electronic cash' after all.

**Comment by dagurval** [comment]
Peer selection for _outgoing connections only_ is needed to be able to use the thin blocks feature of XT. Core 0.12 simply does not support thin blocks. If this is distasteful to you, please disable this feature on your node.

Anyhow, If you look at what peers XT nodes are connected to, you will see that there is none to little segregation of peers in practice. Here is my node: 

> $ ./bitcoinxt/src/bitcoin-cli getpeerinfo | grep subver | wc -l
> 94
> $ ./bitcoinxt/src/bitcoin-cli getpeerinfo | grep subver | grep "Satoshi:0.12" | wc -l
> 36
> 

38% of the nodes my node is connected to is Bitcoin Core 0.12. With a bigger sample size, that number is likely to climb closer to the Core 0.12 share of ~50% of the network nodes.

**Comment by NanoAkron** [comment]
Which feature - connecting only to XThin supporting nodes, or XThin itself?

**Comment by dagurval** [comment]
You cannot disable peer selection without disabling thin blocks at this time.

**Comment by dgenr8** [comment]
Setting ``-use-thin-blocks=3`` (xthin blocks only mode) will keep outbound connections to Unlimited, and soon Classic, nodes that provide xthin service.  It also has the effect of not downloading from peers that don't support xthin.

**Comment by dagurval** [comment]
AFAIK all clients now support either xthin or compact blocks. XT supports both. Low chance of segregating peers with this feature now.
